### ISSUE TITLE
Unable to find icon for domain .... (undefined)

### ISSUE THREAD

**Issue by Mariusthvdb** [home-assistant/frontend#7589]
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

On clicking 'e' or 'c', this is shown in inspector:

<img width="666" alt="Schermafbeelding 2020-11-03 om 12 04 41" src="https://user-images.githubusercontent.com/33354141/98011053-b114f380-1df7-11eb-99b1-7883833d7a5f.png">

## Expected behavior

well, no to have this happen

## Steps to reproduce

A simple click on 'c' or 'e' to call the shortcut bar 

## Environment

- Home Assistant release with the issue: 117.2
- Last working Home Assistant release (if known): 
- Browser and browser version: Safari, Chrome, Firefox
- Operating system: Mac

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

as posted above,
```txt
Unable to find icon for domain ping (undefined)
```
for all domains

**Comment by spacegaier** [comment]
I can confirm that.
@donkawechico Sounds like something for you, since QB related.

**Comment by donkawechico** [comment]
Not as quick-bar related as it seems. This "issue" has been around since before quick bar, it's just that quick bar is particularly good at revealing _all_ of it at once.

Issue being that we have no way of automatically choosing icons for domains, someone has to manually choose icons and add them to the `domainIcon()` method.

I don't mind taking a pass at updating the mappings (gotta do something to take my mind off my election anxiety). But it'll just become an issue again later when new domains get a `reload` service.

**Comment by spacegaier** [comment]
Thanks for the info! 

Although I can see in the OPs screenshots that domain "filesize" is mentioned which does provide a preferred icon via the method `icon()` as do probably many/most other integrations as well. Couldn't we use that (perhaps even with a higher priority than the default domain icons we define in the frontend)?

**Comment by KTibow** [comment]
Should additional domains be added, or not? Are these used as a fallback, or are the entity's native icons better and used by default?

**Comment by donkawechico** [comment]
>Although I can see in the OPs screenshots that domain "filesize" is mentioned which does provide a preferred icon ...

Ah okay, fair point. My local dev doesn't have any reloadable integrations of that type, so this type of icon access never came up. I'll take a look

**Comment by donkawechico** [comment]
Interesting. Turns out it's not at all trivial to find the icon for entities for particular platforms of existing integrations (e.g. 
 the "filesize" platform for `sensor` integration).

In the frontend we can know all of this:
1. There exists a `filesize.reload` service.
2. You have a bunch of entities that have icons assigned by their integration/platform.
3. You have a sensor named "sensor.stats_for_file_name".

The problem is that `sensor.stats_for_file_name` doesn't have any attributes to tell us that it was created by the "filesize" platform. And there's no API in core for finding that out either.

Bottom line is that there isn't a way to see a `filesize.reload` service, and find the _entity_ icon for that platform.

**Comment by Mariusthvdb** [comment]
couldn't the need to find the icon in the first place be prevented then? Why is it there at all?

**Comment by KTibow** [comment]
Should this be moved to Discussions, as these are warnings, not errors?

**Comment by donkawechico** [comment]
> couldn't the need to find the icon in the first place be prevented then?

Yes, and that's the current plan. Or to use icons representing the verb rather than the noun (e.g circle arrow for all reloads)

> fwiw, Ive noticed this before and this was how it was handled

Yep. That's what I was talking about in earlier comments. Manually adding to domain_icon or the const

**Comment by donkawechico** [comment]
> Should this be moved to Discussions, as these are warnings, not errors?

Not sure the protocol there. I personally don't mind keeping it an issue to make sure I remember it for next release.
